Mergin/switch_sources_dialog.py

```diff
@@ -0,0 +1,202 @@
+import os
+import yaml
+from dataclasses import dataclass, field
+import typing
+import re
+
+import psycopg2
+
+from qgis.core import (
+    QgsProject,
+    QgsDataSourceUri,
+    QgsMapLayer,
+    Qgis,
+    QgsVectorLayer,
+)
+from qgis.gui import QgsFileWidget, QgisInterface
+from qgis.PyQt import uic
+from qgis.PyQt.QtWidgets import QWizard, QLineEdit
+
+
+base_dir = os.path.dirname(__file__)
+ui_select_dbsync_page, base_select_dbsync_page = uic.loadUiType(
+    os.path.join(base_dir, "ui", "ui_switch_datasources_select_dbsync.ui")
+)
+ui_select_qgsproject_page, base_select_qgsproject_page = uic.loadUiType(
+    os.path.join(base_dir, "ui", "ui_switch_datasources_select_updated_project.ui")
+)
+
+
+DBSYNC_PAGE = 0
+QGS_PROJECT_PAGE = 1
+
+
+@dataclass
```

Comment:
to me a dataclass is a class encapsulating some simple data - but a connection object isn't quite different IMHO. Could you please change it to ordinary class? (I also feel uneasy that you can currently pass e.g. `db_tables` to its `__init__` constructor...)